numba_rvsdg/core/datastructures/ast_transforms.py

```diff
@@ -490,73 +490,73 @@ def function(a: int) -> None
-                    if i == a:                # check for early exit
-                        i = 420               # set i to some wild value
-                        break                 # early exit break while True
-                else:                         # for-else clause
-         *          i == __iter_last_1__      # restore value of i
-                    c += 1                    # execute code in for-else clause
-                    break                     # regular exit break while True
-            return c, i
+        using the available block and edge primitives of the CFG::
+
+            def function(a: int) -> None:
+                c = 0
+                __iterator_1__ = iter(range(10))  ## setup iterator
+                i = None                          ## assign target, i
+                while True:                       # loop until we break
+                    __iter_last_1__ = i           ## backup value of i
+                    i = next(__iterator_1__, '__sentinel__')  # * get next i
+                    if i != '__sentinel__':       ## regular iteration
+                        c += i                    # add to accumulator
+                        if i == a:                # check for early exit
+                            i = 420               # set i to some wild value
+                            break                 # early exit break while True
+                    else:                         # for-else clause
+                        i == __iter_last_1__      ## restore value of i
+                        c += 1                    # execute code in for-else
+                        break                     # exit break while True
+                return c, i
 
         The above is actually a full Python source reconstruction. In the
         implementation below, it is only necessary to emit some of the special
-        assignments (marked above with a *-prefix above) into the blocks of the
+        assignments (marked above with a #-prefix above) into the blocks of the
         CFG.  All of the control-flow inside the function will be represented
         by the directed edges of the CFG.
 
         The first two assignments are for the pre-header:
 
-         *  __iterator_1__ = iter(range(10))  # setup iterator
-         *  i = None                          # assign target, in this case i
+         *  ``__iterator_1__ = iter(range(10))  ## setup iterator``
+         *  ``i = None                          ## assign target, i``
 
         The next three is for the header, the predicate determines the end of
         the loop.
 
-         *      __iter_last_1__ = i           # backup value of i
-         *      i = next(__iterator_1__, '__sentinel__')  # get next i
-         *      if i != '__sentinel__':       # regular iteration
+         *      ``__iter_last_1__ = i           ## backup value of i``
+         *      ``i = next(__iterator_1__, '__sentinel__')  # * get next i``
```

Comment:
Small nitpick but the rendering of this particular line makes the RHS comment shift a bit more right. Which overall makes it look very haphazard. Could all the RHS comments in this sections be shifted right by the same amount so as to match the indentation of this particular line

Reply:
The indentation is such that it is easier to keep the lines in sync with the above as the indentation level is matched. For the output, they render without all the leading whitespace anyway as shown here. I would prefer thus to keep them as is.

<img width="793" alt="Screenshot 2024-05-29 at 14 02 46" src="https://github.com/numba/numba-rvsdg/assets/41563/696ad51b-40fa-42e7-b9d6-b70220216e93">